src/main/java/org/elastos/hive/vendors/onedrive/OneDriveClient.java

```diff
@@ -59,6 +61,7 @@ public synchronized void login(Authenticator authenticator) throws HiveException
 
 		try {
 			future.get();
+                        authHelper.checkExpired(null);
```

Comment:
Why do we need here to check expired ? Even though, it should be done within authHelper.loginAsync()

Reply:
Yes, you're right.

Reply:
Yes, it's better invoking checkExpired(...) at  loginAsync.